multus_sriov_cni_install.sh

```diff
@@ -0,0 +1,202 @@
+
+export SRIOV_CNI_BRANCH=${SRIOV_CNI_BRANCH:-master}
+export SRIOV_CNI_REPO=${SRIOV_CNI_REPO:-https://github.com/intel/sriov-cni}
+
+export PLUGINS_BRANCH=${PLUGINS_BRANCH:-master}
+export PLUGINS_REPO=${PLUGINS_REPO:-https://github.com/containernetworking/plugins.git}
+
+export SRIOV_NETWORK_DEVICE_PLUGIN_REPO=${SRIOV_NETWORK_DEVICE_PLUGIN_REPO:-https://github.com/intel/sriov-network-device-plugin}
+export SRIOV_NETWORK_DEVICE_PLUGIN_BRANCH=${SRIOV_NETWORK_DEVICE_PLUGIN_BRANCH:-master}
+
+export GOROOT=${GOROOT:-/usr/local/go}
+export GOPATH=${WORKSPACE}
+export PATH=/usr/local/go/bin/:/opt/cni/bin/:$GOPATH/src/k8s.io/kubernetes/third_party/etcd:$PATH
+
+export CNI_BIN_DIR=${CNI_BIN_DIR:-/opt/cni/bin/}
+export CNI_CONF_DIR=${CNI_CONF_DIR:-/etc/cni/net.d/}
+export ALLOW_PRIVILEGED=${ALLOW_PRIVILEGED:-true}
+export NET_PLUGIN=${NET_PLUGIN:-cni}
+
+export KUBE_ENABLE_CLUSTER_DNS=${KUBE_ENABLE_CLUSTER_DNS:-false}
+export API_HOST=$(hostname).$(hostname -y)
+export API_HOST_IP=$(hostname -I | awk '{print $1}')
+export KUBECONFIG=${KUBECONFIG:-/var/run/kubernetes/admin.kubeconfig}
+
+#TODO add autodiscovering
+export MACVLAN_INTERFACE=${MACVLAN_INTERFACE:-eno1}
+export SRIOV_INTERFACE=${SRIOV_INTERFACE:-enp3s0f0}
+export VFS_NUM=${VFS_NUM:-4}
+
+echo "Working under $WORKSPACE folder"
+mkdir -p $WORKSPACE
+mkdir -p $LOGDIR
+mkdir -p $ARTIFACTS
+mkdir -p $CNI_CONF_DIR
+mkdir -p $CNI_BIN_DIR
+
+pushd $WORKSPACE
+
+
+function configure_multus {
+
+    echo "Configure Multus"
+    kubectl create -f $WORKSPACE/multus-cni/images/multus-daemonset.yml
+    sleep 60
+    cat > $CNI_CONF_DIR/00-multus.conf <<EOF
+{
+    "name": "multus-cni-network",
+    "type": "multus",
+    "capabilities": {
+        "portMappings": true
+    },
+    "delegates": [
+        {
+            "cniVersion": "0.3.0",
+            "name": "macvlan-network",
+            "type": "macvlan",
+            "master": "$MACVLAN_INTERFACE",
+            "mode": "bridge",
+            "ipam": {
+                "type": "host-local",
+                "subnet": "192.168.1.0/24",
+                "rangeStart": "192.168.1.100",
+                "rangeEnd": "192.168.1.216",
+                "routes": [{"dst": "0.0.0.0/0"}],
+                "gateway": "192.168.1.1"
+            }
+        }
+    ],
+    "logFile": "$LOGDIR/multus.log",
+    "logLevel": "debug",
+    "kubeconfig": "$KUBECONFIG"
+}
+EOF
+    cp $CNI_CONF_DIR/00-multus.conf $ARTIFACTS
+    return $?
+}
+
+
+function download_cni {
+    if ! $RECLONE ; then
+        return 0
+    fi
+    #trap read debug
+
+    echo "Download $MULTUS_CNI_REPO"
+    git clone $MULTUS_CNI_REPO $WORKSPACE/multus-cni
+    cd $WORKSPACE/multus-cni
+    git checkout $MULTUS_CNI_BRANCH
+    git log -p -1 > $ARTIFACTS/multus-cni.txt
+    cd -
+
+    echo "Download $SRIOV_CNI_REPO"
+    git clone $SRIOV_CNI_REPO $WORKSPACE/sriov-cni
+    pushd $WORKSPACE/sriov-cni
+    git checkout $SRIOV_CNI_BRANCH
+    git log -p -1 > $ARTIFACTS/sriov-cni.txt
```

Comment:
I didn't find where `sriov` cni binary is installed. shall we do it here or deploy it via `sriov-cni-daemonset.yaml`?

Reply:
it would be better to use the daemonset